src/session/session.c

```diff
@@ -953,8 +959,12 @@ main (int argc,
   write_control_string ("challenge", "*");
   write_control_end ();
 
-  /* And get back the authorization header */
+  /* And get back the authorization response; this should not take forever, time-bound it to guard
+   * against hacking/PID recycling attempts */
+  sigaction (SIGALRM, &(struct sigaction) { .sa_handler = on_authorize_timeout, .sa_flags = 0 }, NULL);
+  alarm (60);  /* like cockpit_ws_auth_response_timeout on the ws side */
   char *authorization = read_authorize_response ("authorization");
+  alarm (0);
```

Comment:
I'd be inclined to move the `alarm(0)` down until after we do all the cgroups checking stuff just to drive the point home a bit more.  Maybe the attacker could find another creative way to delay us, otherwise... maybe we innocently read a file that they have on a fuse filesystem which they block indefinitely.... there's quite a lot of tricks these days.

Reply:
I'm a bit nervous about that, like interrupting password/2FA/smart card auth like that, but let's see -- if it's too aggressive, we at least fail towards the "safe"/denied side.